standalone/past_cookies_jar.py

```diff
@@ -0,0 +1,60 @@
+"""
+This script lists all unique cookies seen.
+You may collect them based on the associated domain, you
+just need to change the 'cookie_domain_regex' variable
+below.
+"""
+
+from org.parosproxy.paros.model import Model;
+import re;
+from org.parosproxy.paros.view import AbstractFrame;
+from org.zaproxy.zap.utils import ZapTextArea;
+from javax.swing import JScrollPane;
+
+""" Change this regex to match the desired domain """
+cookie_domain_regex = ".+"
+
+sessionId = Model.getSingleton().getSession();
+tbHist = Model.getSingleton().getDb().getTableHistory();
+
+def collect(msg):
+  """ Collecting cookie from HttpMessage """
+  cookies = msg.getRequestHeader().getHttpCookies();
+  return cookies;
+
+class OutputWindow (AbstractFrame):
+  def __init__(self, text):
+    self.setAlwaysOnTop(False);
+    self.setSize(700, 500);
+    ta = ZapTextArea(text);
+    sp = JScrollPane(ta, 
+		JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED, 
+		JScrollPane.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+    self.add(sp);
+    self.setVisible(True);
+    
+if (tbHist != None):
+  unique_cookies = list();
+  print("Collecting Unique Cookies...");
+  for index in tbHist.getHistoryIds(sessionId.getSessionId()):
+    try:
+      msg = tbHist.read(index).getHttpMessage();
+      results = collect(msg);
+      if results is None:
+        continue;
+      for item in results:
+        item.setDomain(msg.getRequestHeader().getURI().getHost());
+        match = re.search(cookie_domain_regex, item.getDomain(), re.IGNORECASE);
+        if(match):
```

Comment:
This check could be done outside/before the for loop.

Reply:
Ok, i could get the domain out of the loop, but i need to set and match it inside the loop.

Reply:
Does the match also need to be? The host is the same for all the cookies being checked in the loop, so it could be done just once before. Or is that to be easier to change if/when the cookies have other domain than the message?